packages/messenger-widget/src/contexts/Shared.ts

```diff
@@ -23,10 +23,12 @@ const getMainnetProvider = () => {
 export const initialState: GlobalState = {
     connection: {
         defaultServiceUrl: process.env.REACT_APP_BACKEND as string,
+        reset: undefined,
```

Comment:
Why is there a reset property on every object?

Reply:
In the current implementation, when the account changes the context API states do not get reset. So I added a reset action for every context states so that on account change we can use this action and reset entire context storage.

Reply:
Yeah but what is the purpose of having a reset prop on every reducer ?

Reply:
We need to reset data of every reducer as new account is connected. So it is resetted to default state. Since every reducer is created separately so I had to reset all. As we move further and remove legacy state managment code, this will also get removed

Reply:
I understand that we need to reset but why do we need a reset propery that seems to be always set to undefined ?

Reply:
If you see the reducer code, for reset action all reducers are set to their initial state. undefined is used everywhere just to remain consistent. In reset actually we don't need to set specific value so undefined is taken. We just need trigerring of this action no matter what the payload is, so undefined is set.

Reply:
But what would happen if the reset param will be removed entirely ? The reset action would then return the initialState